### ISSUE TITLE
Parallelize the server compute

### ISSUE THREAD

**Issue by glebfann** [apple/swift-homomorphic-encryption#138]
Following up on this [comment](https://github.com/apple/swift-homomorphic-encryption/issues/45#issuecomment-2276367503). This would be really helpful for those trying to scale, as any performance improvements are valuable :)

> Not today, though it should be relatively straightforward to parallelize the server compute in a few places:
[here](https://github.com/apple/swift-homomorphic-encryption/blob/1cff02b950bfac8b572936169ceae8ef45e1af80/Sources/PrivateInformationRetrieval/MulPir.swift#L397-L398) 
[here](https://github.com/apple/swift-homomorphic-encryption/blob/1cff02b950bfac8b572936169ceae8ef45e1af80/Sources/PrivateInformationRetrieval/MulPir.swift#L351-L352)
[here](https://github.com/apple/swift-homomorphic-encryption/blob/1cff02b950bfac8b572936169ceae8ef45e1af80/Sources/PrivateInformationRetrieval/PirUtil.swift#L128)

**Comment by karulont** [comment]
There are multiple ways to scale & parallelize.

If you want to get the lowest latency for a single query, we would need to parallelize the server compute.
However, if you need highest throughput (many queries) parallelizing inside individual queries likely does not help.

**Comment by glebfann** [comment]
Yeah, I meant the first option - reducing latency for a single query. Any performance improvement would be valuable anyway.
Seems like it also could be a good first issue to tackle.

**Comment by fboemer** [comment]
http://github.com/apple/swift-homomorphic-encryption/pull/306 adds an option to parallelize the server compute.
